SimpleITKConfig.cmake.in

```diff
@@ -45,6 +45,9 @@ if(NOT "SimpleITK_VERSION_HASH@" STREQUAL "")
 endif()
 set(SimpleITK_VERSION_COUNT)
 
+if(@SimpleITK_USE_ELASTIX@)
+  set(SimpleITK_USE_ELASTIX ON)
```

Comment:
This should just be `set(SimpleITK_USE_ELASTIX @SimpleITK_USE_ELASTIX@)` so it's always defined.

Then the exlasix configuration file needs to be included if `SimpleITK_USE_ELASTIX` is true and it hasn't been included before. See below  line 65 for ITK.

Reply:
@blowekamp Thanks, can you please elaborate on how or where to include the elastix configuration file? Do you mean "ElastixConfig.cmake"? I'm now including the elastix Targets file, in three CMakeLists files:

    include(${CMAKE_BINARY_DIR}/../Elastix-build/ElastixTargets.cmake)

Is that OK?

Reply:
I was looking for that file in the install directory and couldn't find it there as convention would dictate. There appear to be some best practices not following by Elastix. This is part of the reason I mentioned to defer it.

Referencing the binary directory would prevent the project from being usable if installed.

I would expect it to be something like (capitalization of variables?):
```
if( SimpleITK_USE_ELASTIX and NOT ELASTIX_CONFIG_TARGETS_FILE)
  find_package(Elastix REQUIRED) # may require exact version
endif()
```

Reply:
@blowekamp I understand your point, referencing the binary directory of elastix, instead of its install directory isn't according to the best practices. Would be nice if we could fix that later.

Do you still have a suggestion regarding those visibility warnings, at https://open.cdash.org/viewBuildError.php?type=1&buildid=7827932 ?  Could it be related to the way elastix links to ITK, versus the way SimpleITK links to ITK?   

elastix just does `include( ${ITK_USE_FILE} )`:
https://github.com/SuperElastix/elastix/blob/345a43f615a22c37eb85cb682670c7bb875ad8a2/CMakeLists.txt#L64

Reply:
> @blowekamp I understand your point, referencing the binary directory of elastix, instead of its install directory isn't according to the best practices. Would be nice if we could fix that later.

My suggestion earlier was not to have Elastix in the examples for the initial merge, and come back to these issues.

> 
> Do you still have a suggestion regarding those visibility warnings, at https://open.cdash.org/viewBuildError.php?type=1&buildid=7827932 ? Could it be related to the way elastix links to ITK, versus the way SimpleITK links to ITK?

I suspect that Elastix needs to be compiled with these CMake options:
https://github.com/SimpleITK/SimpleITK/blob/master/SuperBuild/External_ITK.cmake#L81-L83

> 
> elastix just does `include( ${ITK_USE_FILE} )`: https://github.com/SuperElastix/elastix/blob/345a43f615a22c37eb85cb682670c7bb875ad8a2/CMakeLists.txt#L64

Doing this as oppose to directly linking just the libraries needed does introduce over link.